data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -236,18 +309,52 @@ _ "third verse"#enddef
         name = turn end
         first_time_only = no
 
-        [modify_unit]
+        [store_unit]
             [filter]
                 [has_attack]
-                    special_id = once_per_turn_$weapon.name
+                    special_id = once_per_turn
                 [/has_attack]
             [/filter]
 
-            [set_variable]
-                name = $weapon.name|_used
-                value = 0
-            [/set_variable]
-        [/modify_unit]
+            variable = opts_units
+        [/store_unit]
+        [foreach]
+            array = opts_units
+            variable = current_opts_unit
+            [do]
+                [foreach]
+                    array = opts_units
+                    variable = current_opts_unit
```

Comment:
This shadows the outer loop`s `current_opts_unit` and it looks like the outer loop is redundant.